cli/cmds.go

```diff
@@ -87,6 +88,61 @@ func Make() *cobra.Command {
 		},
 	})
 
+	//options : only count, blocks diff/found, total lines diff, etc
+	root.AddCommand(&cobra.Command{
+		Use:   "upgrade012 [file]",
+		Short: "formats terraform blocks to 0.12 format in a single file or on stdin",
+		Args:  cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			filename := ""
+			if len(args) == 1 {
+				filename = args[0]
+			}
+			common.Log.Debugf("terrafmt upgrade012 %s", filename)
+
+			blocksFormatted := 0
+			br := blocks.Reader{
+				LineRead: blocks.ReaderPassthrough,
+				BlockRead: func(br *blocks.Reader, i int, b string) error {
+					var fb string
+					var err error
+					if viper.GetBool("fmtcompat") {
+						fb, err = upgrade012.FmtVerbBlock(b)
+					} else {
+						fb, err = upgrade012.Block(b)
+					}
+
+					if err != nil {
+						return err
+					}
+
+					_, err = br.Writer.Write([]byte(fb))
+
+					if err == nil && fb != b {
+						blocksFormatted++
+					}
+
+					return nil
+				},
+			}
+			err := br.DoTheThing(filename)
+
+			fc := "magenta"
+			if blocksFormatted > 0 {
+				fc = "lightMagenta"
+			}
+
+			if viper.GetBool("verbose") {
+				// nolint staticcheck
+				fmt.Fprintf(os.Stderr, c.Sprintf("<%s>%s</>: <cyan>%d</> lines & formatted <yellow>%d</>/<yellow>%d</> blocks!\n", fc, br.FileName, br.LineCount, blocksFormatted, br.BlockCount))
+			}
+			if err != nil {
+				return err
+			}
+			return nil
```

Comment:
could we add the new check for the diff function here were it does an os.exit(-1) if there are any blocks with error?